src/Http/Http.Abstractions/src/Extensions/UsePathBaseExtensions.cs

```diff
@@ -31,6 +31,30 @@ public static IApplicationBuilder UsePathBase(this IApplicationBuilder app, Path
             return app;
         }
 
+        const string globalRouteBuilderKey = "__GlobalEndpointRouteBuilder";
+        const string useRoutingKey = "__UseRouting";
```

Comment:
Can we get rid of `"__GlobalEndpointRouteBuilder"` and use `"__UseRouting"` for everyting instead? `builder.Properties[globalRouteBuilderKey]` is just a marker and nothing actually uses the value, right?

Reply:
It's still being used to preserve the endpoints from the original routing pass.
https://github.com/dotnet/aspnetcore/pull/42876/files#diff-7671a33543dabeffdd1b348bc37fb748c2b8c3713ad8dc7b5f727af6b28e30bbR45

Reply:
How about adding both as a tuple or something and then using the `Func<IApplicationBuilder, IApplicationBuilder>` in . I don't like using different patterns in `UseExceptionHandler()`, `UseRewriter()`, etc... We should make a shared source internal extension method and call that everywhere.

Really, we should just add new public API like a `IReroutableApplicationBuilder.Reroute()` or something that we can expose from Http.Abstractions, but we can skip that for this PR.